### PR TITLE
Whew boy a lot of molitz b QoL 

### PR DESCRIPTION
**Pull request by ThatFiveGuys**
## About the PR 
So uhhh, theres a molitz b canister in code, cant make it though. Im not gonna put too much work into this because idk I already rewrote this thing like 3 times already and yeah. Uhhh the amount of times it can make gas has been increased and now sound. Oh also explosions increase the iterations of gas production it can undergo. Yeah.
(Cool sparkles and gas sound)

https://user-images.githubusercontent.com/73148980/120946417-0accfb80-c702-11eb-93fe-1211197d55ee.mp4

## Why's this needed? 
Sense the removal of Agent B from biodome, hellburns have been practically impossible to do, this pr should change that fact. Currently the crystals are near impossible to use, producing way too much oxygen for way too little time, at freezing temperatures. Also cool molitz b sound.

## Changelog

```changelog
(u)Ikea
(*)Molitz B has more distinct indicators for if it's producing agent b and oxygen or just oxygen. Explosions near crystals leads to resonance, increasing the total amount of times Molitz can produce gas. Molitz B gas output has been overall buffed. Amount of Molitz B obtainable from mining has been nerfed.
```

### CODE REVIEW

**Comment by Tarmunora** [comment]
Would rather see the gas yield from the material increased rather than just printing canisters

File: code/obj/machinery/manufacturer.dm
Comment (Azrun) [inline code/obj/machinery/manufacturer.dm:2378]: Maybe I just don't really get arguments as to why, but this still doesn't quite make sense to me as someone doesn't actually play on RP.  The mechanical difference doesn't really benefit the feel/flow/behavior of the game mode as other deviations.  Would the fact that it is a hidden item mean it is "forbidden" and/or "meta knowledge" and thus push it to antagonistic use?  Time for audits! 🧑‍💼 

Ideally the amount of Agent B you get from the raw materials is <= what could be harvested from the material to justify simplification of the process.

**Comment by Awfullyfishy** [comment]
Not particularly happy about excluding RP. They already don't get to play with Agent B much due to the weirdness of rushing azones as an Engineer, this would promote Mining and Engineering cooperation and give people opportunities to play with it and toy with hellburns again. Of course, I agree with the reasoning behind restricting it due to the danger of boiling the station, but I'd prefer the option of hellburns existing for experienced people. I believe it being a hidden recipe and requiring Mining is more than sufficient to weed out the people unprepared for hellburns, and the ones that make mistakes and ruin the round, well, singuloose is a thing on RP and I see no reason why this should be any different.

**Comment by nefarious6th** [comment]
can I ask for a [wiki] tag on this?
Ty

Unrelated and about pr contents:
TBH I would not fully trust that RP engineers would respect the experience of others on the station to not cause issues with massive burns; simply because I've seen this happen recently with the Sing too, and with the TEG a lot before many of the changes. Though, I would much rather codify that particular flavor of engine negligence and murder and being against the rules and try to culturally make RP engineers more conscious and responsible versus trying to attempt that by mechanically locking them from some aspects of the game. Though I suppose there's something to be thought on that Agent B is only absolutely required for large burns that will inevitably either require deconstruction of the station or will harm other crew. It could be nice for experimentation but it brings up the point of to what end? When someone finally figures out how to use it, they will maximize their burn, and the problems will start. I wouldn't really want to see this on RP because of that but I wonder if there is a better way? Maybe making it something more rules-adjacent and expecting people to follow it in good faith is too trusting, though.

**Comment by penny64** [comment]
It's really not acceptable to if def this kind of thing for RP or not, I know it's really looked down at it's not good to fracture gameplay between the playerbases except where already established (almost just antags)

**Comment by ThatFiveGuys** [comment]
> Additionally, it makes absolutely no sense to leave RP in the situation where engineers who will increasingly know about agent B will leave to get

If someone is azone rushing on rp, ahelp it please.

**Comment by ThatFiveGuys** [comment]
> Would rather see the gas yield from the material increased rather than just printing canisters

Personally Id dislike the latter, since azone rushing would still be the best way to get gas. If we were to remove the tank, this would make canbombs practically impossible, unless you make it to where 7 molitz b can last a burn for 40 minutes. Canbombs are already way harder to do, taking a full round of dedication and around 50 minutes of work. However I do agree that the gas yield should be increased, currently it's way too low. Ideally Id like it if toxins was about manually collecting molitz B while engineering was about fabricating it.

**Comment by ThatFiveGuys** [comment]
> can I ask for a [wiki] tag on this?

While Agent B should be added to wiki, may I make a request? Id prefer it if what agent B was left purposefully vague and the wiki boil down to it exists (perhaps maybe some ic shit about how this gas is experimental and its properties arent well understood). The properties of agent B shouldnt be too hard to discern through experimentation, and having a new player decide to add it to a burn only to realize holy shit this is getting fuckin hot wow, versus just being told what it does seems like the former would be better.

File: code/obj/machinery/manufacturer.dm
Comment (ThatFiveGuys) [reply code/obj/machinery/manufacturer.dm:2378]: It being an hidden item doesnt really stop people from using it, considering other hidden items are practically required for effective job use (see botanist hacking fabs for weed, or borgs hacking fabs for expansion upgrades) (Note I think this is fine tbh).

**Comment by ThatFiveGuys** [comment]
> well, singuloose is a thing on RP and I see no reason why this should be any different.

While I'll likely change it to available on RP too there in a bit, due to most people being in favor, this is a common point I saw brought up and Id like to address. The thing about singularities is 90% of the time only 2 people die, they get shot down, and besides a bit more pressure on cargo to get cash to repair it, the station goes on as normal. You cannot repair a hellburn, at least practically. You can stop a hellburn before it becomes an issue, but once half the station gets superheated, good luck cooling that down. The thing that pushed me over the edge and prompted me to make it non RP only (was already seriously considering this beforehand) was someone azone rushing and doing a hellburn on RP, and they didn't even vent. Had they done the burn properly (which wouldnt be hard for them to do) the fact that they didnt vent would lead to a great deal of suffering, and likely ruin that round for many people due to the abrupt nature of hellburns gone wrong. If hellburns didn't practically require azone rushing, theyd be far more common on RP. While loose hellburns would likely be less common then pre changes on RP, I fear that these changes would still make them semi common on RP. If my fears are proven true I'll likely make an additional pr in the future that reinstates the RP mode locking.

File: code/modules/atmospherics/portable/canister.dm
Comment (ThatFiveGuys) [reply code/modules/atmospherics/portable/canister.dm]: Now 5% of a normal canister mol wise and cost increased, want your feedback for changing amount produced by molitz b. Marking as resolved for now

**Comment by nefarious6th** [comment]
> > can I ask for a [wiki] tag on this?
> 
> While Agent B should be added to wiki, may I make a request? Id prefer it if what agent B did was left purposefully vague and the wiki boil down to it exists (perhaps maybe some ic shit about how this gas is experimental and its properties arent well understood). The properties of agent B shouldnt be too hard to discern through experimentation, and having a new player decide to add it to a burn only to realize holy shit this is getting fuckin hot wow, versus just being told what it makes me feel like the former would be better.

I just want a tag on the creation of it though the properties of agent b in general ultimately arent secret

File: code/modules/atmospherics/portable/portable_atmospherics.dm
Comment (ThatFiveGuys) [reply code/modules/atmospherics/portable/portable_atmospherics.dm:112]: Most practical cleanest way I could find, moving it to that would lead to code bloat, resolving this for now_

Comment (Azrun) [reply code/modules/atmospherics/portable/portable_atmospherics.dm:112]: Parents don't need to care about thier children.  Specific behavior should preferable be handled by the child with `..()` to allow parent to handle shared behavior.

**Comment by Azrun** [comment]
> the properties of agent b in general ultimately arent secret

Agreed. I thought the note in the briefcase gives a pretty good description.

**Comment by ThatFiveGuys** [comment]
Id say about the same amount of detail as the note, while the properties aren't secret, there's the issue of giving too much information away on the wiki that can often be better left to experimentation to figure out. Like for example farts on the wiki doesnt state it purges co2. idk your call.

**Comment by zjdtmkhzt** [comment]
IMO having this be a secret recipe is kind of weird, seeing as it's a pretty major part of engineering for a machine that is basically only in engineering.
(In the same way, I think the botany seed fabricators having to be hacked is also kind of dumb.)

Also, yes, I would prefer not having this be unavailable on RP, I think if people are being bad with it, that can just be dealt with on a case-by-case basis, the same as bombings and stuff, that's what rules are for.

**Comment by ThatFiveGuys** [comment]
> IMO having this be a secret recipe is kind of weird, seeing as it's a pretty major part of engineering for a machine that is basically only in engineering.
> (In the same way, I think the botany seed fabricators having to be hacked is also kind of dumb.)
> 
> Also, yes, I would prefer not having this be unavailable on RP, I think if people are being bad with it, that can just be dealt with on a case-by-case basis, the same as bombings and stuff, that's what rules are for.

Only reason why its secret is for lore reasons, agent B isnt a standard gas NT uses, its an experimental gas they barely understand., having it be just on there wouldnt really make sense. Engineers have hacking tools on them, its not hard for them to hack a fab.

**Comment by ThatFiveGuys** [comment]
Note about this pr: The feature creep of this pr has expanded heavily, it has become less of "oh now you can make canisters" and more of hellburn rerework. The desc and title will stay the same for a bit, however that wont be permanent (aka once I finish coding and iron out the kinks Ill get to changing that info). The RP mode thing will change, just keeping that for last.

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (Azrun) [inline code/modules/materials/Mat_MaterialProcs.dm:410]: Assuming my math is correct... 100 iterations to deplete generating 19.099 Agent B and 100 Oxygen and will take 110 iterations to hit minimum react on reaction_rate() (7.33 minutes)

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (Azrun) [inline code/modules/materials/Mat_MaterialProcs.dm:393]: Add comment justifying behavior

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (Azrun) [inline code/modules/materials/Mat_MaterialProcs.dm:420]: 😕, would it make more sense if there was no total_oxygen and this defined the number of iterations? Only iterations 91-99 will have < 0.2 moles of Agent B added.

**Review by Azrun — COMMENTED** [review]
Feel free to ignore depending on how this is being re-scoped....

**Comment by Azrun** [comment]
Hacked & RP:
Alternatively the note in the RD's briefcase the could be replaced with a dossier that provides the note and a blueprint for a fab recipe.  Unfortunately that does put a LOT of weight on that object.

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (ThatFiveGuys) [reply code/modules/materials/Mat_MaterialProcs.dm:420]: Mostly just lazy code, should be changed yeah

File: code/datums/manufacturing.dm
Comment (ThatFiveGuys) [reply code/datums/manufacturing.dm]: Done mostly

File: code/obj/machinery/manufacturer.dm
Comment (ThatFiveGuys) [reply code/obj/machinery/manufacturer.dm]: Changed

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (ThatFiveGuys) [reply code/modules/materials/Mat_MaterialProcs.dm]: Updated comment accordingly

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (ThatFiveGuys) [reply code/modules/materials/Mat_MaterialProcs.dm:419]: Added a comment to show

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (ThatFiveGuys) [reply code/modules/materials/Mat_MaterialProcs.dm]: Oh yeah I did this comment!

**Comment by lolJamesR** [comment]
i never really use github really, but i personally think that if this gets added, some maps would need a gas extractor added to toxins,  i dont know about you, but the hassle of running to engi and convincing them to let you in is just miserable to me, of course, this is debatable, but its just what i think tbh

**Comment by ThatFiveGuys** [comment]
> i never really use github really, but i personally think that if this gets added, some maps would need a gas extractor added to toxins, i dont know about you, but the hassle of running to engi and convincing them to let you in is just miserable to me, of course, this is debatable, but its just what i think tbh. Also i think that the material cost for the molitz should be lowered a bit, i really think that in fact,

Ive yet to readjust description and name appropiately, but the crystals have been buffed. You should be able to make a maxcap canbomb with the briefcase. High cost of canister for molitz b is intentional, I want it to cost more then what the briefcase can provide. Having engineers rush the rds office for molitz b crystals instead of rushing telesci isnt really a fix to the issues we had previously.

**Comment by ThatFiveGuys** [comment]
Okay this code is now done with new features and works! So yeah there is that.

**Review by Tarmunora — CHANGES_REQUESTED** [review]
When I said "maybe buff crystals instead", I was imagining something more along the lines of a 50-150% increase kind of thing along with maybe a more involved method of getting higher efficiency out of them, not a 100-fold increase in total mols paired with changes to make using them lower-effort (payload temperature change). Also this canister is still >60x stronger than old crystals.

File: code/datums/manufacturing.dm
Comment (Azrun) [inline code/datums/manufacturing.dm:579]:
```suggestion

/datum/manufacture/agentb_can
	name = "Oxygen Agent B Canister"
	item_paths = list("MET-2","molitz_b","uqill")
	item_names = list("Sturdy Metal","Molitz Beta","Uqill")
	item_amounts = list(3,100,5)
	item_outputs = list(/obj/machinery/portable_atmospherics/canister/agent_b)
	time = 100 SECONDS
	create = 1
	category = "Machinery"
	
```

File: code/modules/atmospherics/portable/canister.dm
Comment (Azrun) [inline code/modules/atmospherics/portable/canister.dm:66]:
```suggestion

/obj/machinery/portable_atmospherics/canister/agent_b
	name = "Canister \[Oxygen Agent B\]"
	icon_state = "bluish"
	casecolor = "bluish"
	desc = "A container which holds a large amount of the labelled gas. It's possible to transfer the gas to a pipe system, or the air."
	volume = 500
	
```

File: code/modules/atmospherics/portable/canister.dm
Comment (Azrun) [inline code/modules/atmospherics/portable/canister.dm:137]:
```suggestion
/obj/machinery/portable_atmospherics/canister/agent_b/update_icon()
	if (src.destroyed)
```

File: code/modules/atmospherics/portable/canister.dm
Comment (Azrun) [inline code/modules/atmospherics/portable/canister.dm:743]: Magic number bad! Rawr!

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (Azrun) [inline code/modules/materials/Mat_MaterialProcs.dm:410]: .....

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (Azrun) [inline code/modules/materials/Mat_MaterialProcs.dm:404]: You don't need to assign this to temp.

Is there anything significant about the change of heat by 10000?

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (Azrun) [inline code/modules/materials/Mat_MaterialProcs.dm:415]: Can you add an demo/example of this to the PR?

File: code/obj/machinery/manufacturer.dm
Comment (Azrun) [inline code/obj/machinery/manufacturer.dm:2377]:
```suggestion
	hidden = list(/datum/manufacture/agentb_can)
	
/obj/machinery/manufacturer/mechanic // a blank manufacturer for mechanics
```

File: code/modules/atmospherics/portable/canister.dm
Comment (Azrun) [inline code/modules/atmospherics/portable/canister.dm:66]: Adjust `filled` to get the amount you want instead of dividing it by 8 below.

**Review by virvatuli-gh — CHANGES_REQUESTED** [review]
Audio files sound good but `leakagentb.ogg` needs to be converted to mono, and both files could stand to be compressed a bit.

**Review by ThatFiveGuys — COMMENTED** [review]
Wait fuck this is a misclick and I dont know how to stop doing this Im just going to have this and like approve the changes (Edit: Oh that worked)

File: code/modules/materials/Mat_MaterialProcs.dm
Comment (ThatFiveGuys) [reply code/modules/materials/Mat_MaterialProcs.dm:411]: Done! (Added audio showoff too because thats important)

**Review by virvatuli-gh — APPROVED** [review]

**Comment by Sovexe** [comment]
what is the status of this

**Comment by ThatFiveGuys** [comment]
This pr is still being worked on and not ready to merge. Several features are buggy right now (aka i havent gotten explosion reaction to work yet). Currently working on a related but seperate pr, my plans for this pr will be to comment out fabricator recipe and nerf crystals yet again, likely agent b output per iteration be will equal to total agent b gas output (after molitz b properly works with explosions of course).

**Comment by ThatFiveGuys** [comment]
Wait what the fuck why did mapping happen???

**Comment by ThatFiveGuys** [comment]
Mordent in case you read this, you dont actually have to review this pr, I did a fucky wucky and git treated it as if I changed all files, that included mapping files so it automatically requested you. So uhh yeah.

**Review by Tarmunora — APPROVED** [review]

**Comment by ThatFiveGuys** [comment]
Ill delete this once I fix explosions but dont merge quiteee yet. Nearly finished just need to fix something

**Comment by ThatFiveGuys** [comment]
Whew this pr is finally done ish?? I mean it works fine. Yeah. Feel free to merge at this point. Big big thanks to all the people who helped (Azrun and The Butter Man and Yass are a few that come to mind but other people were great help too!). So uhh yeah, a few things Id like to add in a future pr but arent critical is like. Tying iterations up from resonance to matsci and a mat (mats?) specifically for upping iterations. Also a unique sound to aurally indicate that the amount of iterations has been increased would be useful. So uhh yeah there's the part where we are.

**Comment by Azrun** [comment]
Should the Oxygen Agent B Canister changes be removed? It was removed from the manufacturer.

**Comment by Tarmunora** [comment]
> 
> 
> Should the Oxygen Agent B Canister changes be removed? It was removed from the manufacturer.

Yeah, canister code should probably be split out

**Comment by ThatFiveGuys** [comment]
I mean i can see the use for testing/sprite work being used for azones in the future. No big deal if all code gets removed though.

**Pull request merged**
